### PR TITLE
Ensure 'session._transaction' is cleared after 'run_in_transaction' succeeds

### LINKED ISSUE #3434: Calling run_in_transaction twice causes the second call to fail
It seems that the `_transaction` object in Session is not being reset once the transaction commits. This leads to a "Transaction is already committed" error. Here is a simple code to demonstrate this:

```python
def run_txn():
    from google.cloud import spanner
    spanner_client = spanner.Client()
    instance_id = 'your-instance'
    instance = spanner_client.instance(instance_id)
    database_id = 'example-db'
    database = instance.database(database_id)
    def f(t):
      t.insert_or_update(
          table="Singers",
          columns=('SingerId', 'FirstName'),
          values = [(6, "A"),])
    database.run_in_transaction(f)
    database.run_in_transaction(f)

if __name__ == '__main__':
    run_txn()

```

The error that is raised is:

```python
Traceback (most recent call last):
  File "txn.py", line 17, in <module>
    run_txn()
  File "txn.py", line 14, in run_txn
    database.run_in_transaction(f)
  File "/usr/local/google/home/vikask/venvs/pythonq/local/lib/python2.7/site-packages/google/cloud/spanner/database.py", line 379, in run_in_transaction
    return session.run_in_transaction(func, *args, **kw)
  File "/usr/local/google/home/vikask/venvs/pythonq/local/lib/python2.7/site-packages/google/cloud/spanner/session.py", line 309, in run_in_transaction
    txn.commit()
  File "/usr/local/google/home/vikask/venvs/pythonq/local/lib/python2.7/site-packages/google/cloud/spanner/transaction.py", line 104, in commit
    self._check_state()
  File "/usr/local/google/home/vikask/venvs/pythonq/local/lib/python2.7/site-packages/google/cloud/spanner/transaction.py", line 46, in _check_state
    raise ValueError("Transaction is already committed")
ValueError: Transaction is already committed
```

### PR DESCRIPTION
**Pull request by tseaver**
Closes #3434.

### CODE REVIEW

**Comment by tseaver** [comment]
New system test provoked the same traceback as reported in #3434, and passes with the changed code.

**Comment by tseaver** [comment]
I'll tag `spanner-0.24.2` after merging this to `master`.  Here are the draft release notes:

## google-cloud-spanner-0.24.2

- Correctly marshall `float("+inf")`/`float("-inf")` values to JSON (issue #3016).
- Ensure `session._transaction` is cleared after `run_in_transaction` succeeds. (issue #3434).

File: spanner/tests/system/test_system.py [hunk 1]
```diff
@@ -323,6 +323,24 @@ def _unit_of_work(transaction, test):
         rows = list(self._db.execute_sql(self.SQL))
         self._check_row_data(rows)
 
+    def test_db_run_in_transaction_twice(self):
+        # See issue #3434.
```
Comment (dhermes) [inline spanner/tests/system/test_system.py]: I am :-1: on this comment in the source (a `git blame` accomplishes similar things)

File: spanner/tests/unit/test_session.py [hunk 2]
```diff
@@ -503,6 +503,7 @@ def unit_of_work(txn, *args, **kw):
             unit_of_work, 'abc', some_arg='def')
 
         self.assertEqual(committed, now)
+        self.assertIsNone(session._transaction)
```
Comment (dhermes) [inline spanner/tests/unit/test_session.py]: :+1:

**Pull request merged**